### PR TITLE
fix some oss-fuzz

### PR DESCRIPTION
**Pull request by chenhuitao**
fix some oss-fuzz: 10598, 11112, 11186, 11589, 10674, 10878, 11655, 19610, 19848, 19849, 19850, 19851, 19852.

### CODE REVIEW

File: qemu/fpu/softfloat.c [hunk 1]
```diff
@@ -130,7 +130,7 @@ static int32 roundAndPackInt32( flag zSign, uint64_t absZ STATUS_PARAM)
     absZ = ( absZ + roundIncrement )>>7;
     absZ &= ~ ( ( ( roundBits ^ 0x40 ) == 0 ) & roundNearestEven );
     z = (int32_t)absZ;
-    if ( zSign ) z = - z;
+    if ( zSign && (z != 0x80000000)) z = - z;
```
Comment (aquynh) [inline qemu/fpu/softfloat.c]: where is this 0x80000000 from?

Comment (chenhuitao) [reply qemu/fpu/softfloat.c]: 0x80000000 is MIN_INT32.

File: qemu/target-arm/translate.c [hunk 2]
```diff
@@ -807,7 +807,9 @@ void arm_gen_test_cc(TCGContext *tcg_ctx, int cc, int label)
         break;
     default:
         fprintf(stderr, "Bad condition code 0x%x\n", cc);
```
Comment (aquynh) [inline qemu/target-arm/translate.c]: pls comment out this fprintf line

File: qemu/target-i386/int_helper.c [hunk 3]
```diff
@@ -352,7 +352,7 @@ static int idiv64(uint64_t *plow, uint64_t *phigh, int64_t b)
         neg128(plow, phigh);
     }
     sb = (b < 0);
-    if (sb) {
+    if (sb && (b != 0x8000000000000000LL)) {
```
Comment (aquynh) [inline qemu/target-i386/int_helper.c]: this looks like a trick here. sorry but i feel it is ugly to fix a bug this way. any ideas why oss-fuzzer reports this issue?

File: qemu/target-arm/translate.c (hunk 2 above)
Comment (chenhuitao) [reply qemu/target-arm/translate.c]: O.K.

File: qemu/target-i386/int_helper.c (hunk 3 above)
Comment (chenhuitao) [reply qemu/target-i386/int_helper.c]: The min int8_t is -128(0x80), the max is 127(0x7f), the abs value of min is bigger than max and overflowed. In most machine and compiler, -(-128) = ~(-128) + 1 = -128. Same ruler for other int types. I think the oss-fuzz just report this bug is not good compatible, just like left shift the negative int. This code change keeps the original logic and becomes compatible.

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**